### ISSUE TITLE
[Bug]: Restoring state on a cropped image shifts all layers

### ISSUE THREAD

**Issue by saif-ellafi** [hm21/pro_image_editor#292]
### Package Version

6.0.0

### Flutter Version

3.27.1

### Platforms

Android, iOS, Web, Windows, macOS, Linux

### How to reproduce?

1. Set some layers on an image
2. Crop the image
3. Export state 
4. Load state
5. Notice all layers will have shifted from where they were, relative to the image, and became smaller
6. Save
7. Load state again
8. Notice all layers will have shifted even further, and became even smaller
9. repeat...

<details>
<summary>Show Images</summary>

## Set layers
![image](https://github.com/user-attachments/assets/037d6296-3323-4bab-a348-3f3eaafbe97d)

## Crop image around layers
![image](https://github.com/user-attachments/assets/eb0d07a4-be77-47a5-8139-19b4115f9802)

## Press "OK" - notice already a shift (but we can reposition manually, at this point
![image](https://github.com/user-attachments/assets/85117562-885a-493d-b464-2a30bae2ae6f)
![image](https://github.com/user-attachments/assets/0d63dd25-3c0e-442e-927c-ea04a8e3cf4f)

## Restore state
![image](https://github.com/user-attachments/assets/15095e33-dc8c-4553-b102-afc1f8f36f2a)

</details>

**Comment by hm21** [comment]
Small update:
I am working on a bug fix in [this branch](https://github.com/hm21/pro_image_editor/tree/bugfix/crop-rotate-export). 
The offset issue was caused by the editor exporting the last rendered image size instead of the original rendered image size. However, I also discovered a second issue: when the image fit factor (e.g., "fit to width") changes, the layer shifts and develops an offset in any direction.

**Comment by KrishnaTridhya** [comment]
Hey! Did you find any solution for the above issue, as I am facing the same issue while using the custom bottom widget while using Paint on Image?

| Image 1 | Image 2 |
|---------|---------|
| ![Image](https://github.com/user-attachments/assets/2c90c548-e979-4319-9934-9af396c6d423) | ![Image](https://github.com/user-attachments/assets/681618e1-f893-4af9-8369-ae42c1e539c2) |

**Comment by hm21** [comment]
Hi @KrishnaTridhya, 

Sorry for the late response.

As mentioned in my [comment here](https://github.com/hm21/pro_image_editor/issues/292#issuecomment-2633525895), I am already working on this issue. However, as long as the issue remains open, it still persists.

Based on your images, I assume the problem is not related to the import/export functionality. Instead, it looks like the layers shift after closing the paint editor, right?

**Comment by KrishnaTridhya** [comment]
> Hi [@KrishnaTridhya](https://github.com/KrishnaTridhya),
> 
> Sorry for the late response.
> 
> As mentioned in my [comment here](https://github.com/hm21/pro_image_editor/issues/292#issuecomment-2633525895), I am already working on this issue. However, as long as the issue remains open, it still persists.
> 
> Based on your images, I assume the problem is not related to the import/export functionality. Instead, it looks like the layers shift after closing the paint editor, right?

yes, while configuring custom MainEditorConfigs and PaintEditorConfigs.

**Comment by hm21** [comment]
@KrishnaTridhya Okay, got it. 
Since this is not related to the current issue, which involves importing/exporting state history, I’d appreciate it if you could create a new Q&A in the Discuss tab [here](https://github.com/hm21/pro_image_editor/discussions/new?category=q-a) and post your current editor code. That way, I can review it and help identify the issue.

**Comment by hm21** [comment]
Resolved in #396
